HIP/hip-1037.md

```diff
@@ -0,0 +1,95 @@
+---
+hip: 1037
+title: Protocol Buffer API Specification
+author: Joseph Sinclair <@jsync-swirlds>
+requested-by: Richard Bair <@rbair>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Review
+created: 2024-08-09
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/1037
+updated: 2024-09-18
+---
+
+## Abstract
+The current protocol buffer API definitions are technically correct, but lack
+usable or useful specification. We should add, in comments compatible with
+protoc-gen-doc, specification text that clearly states both requirement and
+expectation for every message, file, and field. This documentation should be
+automatically transformed to markdown documentation that is published to the
+Hedera API site. In the ideal case this specification will be sufficient for
+any qualified team to implement a completely independent consensus node that
+is fully interoperable with the current consensus node software.
+
+## Motivation
+The existing protocol buffer API documentation is brief, entirely descriptive,
+and often inaccurate. This produces two negative consequences. First, it is not
+possible to produce a compatible independent implementation of the Hedera
+consensus node software without extensive reference to the existing source
+code. Second, developers are often unclear how best to interact with the
+Hedera API, and independent SDK developers, in particular, may struggle to
+correctly implement these API calls.
+As part of expanding the Hedera ecosystem we MUST reduce the difficulty and
+hurdles for independent developers when building systems that interact with
+the Hedera network. Clear and accurate API specifications are one element
+required to accomplish this goal.
+
+## Rationale
+This HIP improves the developer experience and enhances decentralization by
+making it easier to design distributed applications or alternative
+implementations of core network systems.
+The use of a markdown format that is automatically generated from the
+functional protocol buffer definition files encourages more frequent updates
+that are more closely tied to the functional updates to the API documents, and
+reduces the inevitable "drift" between functional API and API specification.
+
+## User stories
```

Comment:
We have an example now of a protobuf which is used to talk to a system contract method (for [HIP-632's `isAuthorized` method](https://hips.hedera.com/hip/hip-632#isauthorizedaddress-message-signatureblob-function-usage)) and so that is potentially another user story to mention here: smart contract developers who need to send a protobuf to a smart contract in a transaction.
* This is slightly different from the current "dAPP developer" which is speaking of "Hedera APIs" i.e., HAPI.  Though that user story could be expanded (just a bit) to take care of this case.

Reply:
I am not clear on the use case.  Would you be willing to write a suggested text for this?

Reply:
Does something like this work @david-bakin-sl and @jsync-swirlds ?
```suggestion
## User stories
- As a dApp developer that receives bytes or expected to parse bytes representing HAPI protobuf message I want to have a clear description of the structure and implications of the corresponding proto messages to understand how to correctly parse them. 
```

Reply:
Yes, that is the use case I was thinking of, TY

Reply:
Expanded slightly and added.